packages/plugin-ext-headless/src/main/node/env-main.ts

```diff
@@ -0,0 +1,44 @@
+// *****************************************************************************
+// Copyright (C) 2018 Red Hat, Inc. and others.
+//
+// This program and the accompanying materials are made available under the
+// terms of the Eclipse Public License v. 2.0 which is available at
+// http://www.eclipse.org/legal/epl-2.0.
+//
+// This Source Code may also be made available under the following Secondary
+// Licenses when the conditions for such availability set forth in the Eclipse
+// Public License v. 2.0 are satisfied: GNU General Public License, version 2
+// with the GNU Classpath Exception which is available at
+// https://www.gnu.org/software/classpath/license.html.
+//
+// SPDX-License-Identifier: EPL-2.0 OR GPL-2.0-only WITH Classpath-exception-2.0
+// *****************************************************************************
+
+import { interfaces } from '@theia/core/shared/inversify';
+import { EnvVariablesServer } from '@theia/core/lib/common/env-variables';
+import { RPCProtocol } from '@theia/plugin-ext/lib/common/rpc-protocol';
+import { EnvMain } from '@theia/plugin-ext/lib/common/plugin-api-rpc';
+import { isWindows, isOSX } from '@theia/core';
+import { OperatingSystem } from '@theia/plugin-ext/lib/plugin/types-impl';
+
+export class EnvMainImpl implements EnvMain {
```

Comment:
This seems to be a 1:1 copy from plugin-ext, could we not simply re-use their implementation or at least inherit from it?

Reply:
I think we can do better and delete it; see my response to an earlier comment 😀

Reply:
Actually, now that we are using DI in the plugin hosts, third-party extensions will be able to inject the `EnvExtImpl` into their own ext-classes, so I need to retain it and the main-side implementation as this will be a way to get environment variables.

Reply:
Noted.